### PR TITLE
Fix Horovod distributed backend to set the root_gpu property

### PR DESCRIPTION
**Pull request by tgaddair**
It was discovered that when restoring a model from a checkpoint (`trainer.restore(path, on_gpu=True)`) then performing testing, the model will not be placed on the correct GPU device, because `root_gpu` was not set when using the Horovod backend.

### CODE REVIEW

File: pytorch_lightning/trainer/distrib_parts.py [hunk 1]
```diff
@@ -570,8 +570,9 @@ def horovod_train(self, model):
 
         if torch.cuda.is_available() and self.on_gpu:
             # Horovod: pin GPU to local rank
-            torch.cuda.set_device(hvd.local_rank())
-            model.cuda(hvd.local_rank())
+            self.root_gpu = hvd.local_rank()
```
Comment (Borda) [inline pytorch_lightning/trainer/distrib_parts.py:573]: shall it be rather protected `self._root_gpu` also add it to Trainer.__init__

Comment (tgaddair) [reply pytorch_lightning/trainer/distrib_parts.py]: Moved the logic to the init method.  Seems there isn't a protected `_root_gpu`.  I was using the existing public `root_gpu` property set elsewhere, following this pattern: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/distrib_data_parallel.py#L348

Let me know if the new approach looks good.

**Comment by Borda** [comment]
I saw it also with other PRs, no idea what changed...
`AssertionError: This model is expected to get > 0.5 in test set (it got 0.28125)`

**Comment by tgaddair** [comment]
> I saw it also with other PRs, no idea what changed...
> `AssertionError: This model is expected to get > 0.5 in test set (it got 0.28125)`

Seems this is a problem on master: https://circleci.com/gh/PyTorchLightning/pytorch-lightning/24909?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by tgaddair** [comment]
> > I saw it also with other PRs, no idea what changed...
> > `AssertionError: This model is expected to get > 0.5 in test set (it got 0.28125)`
> 
> Seems this is a problem on master: https://circleci.com/gh/PyTorchLightning/pytorch-lightning/24909?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Looking at the build history, it looks like #1498 was when the failure started occurring.  Maybe @williamFalcon knows more.

**Comment by Borda** [comment]
> Looking at the build history, it looks like #1498 was when the failure started occurring. Maybe @williamFalcon knows more.

I agree that it is not linked to this PR, but your mentioned PRs seems that did not touch it either and its last tests were passing...
Do we even need this test `test_lbfgs_cpu_model` ?

**Comment by williamFalcon** [comment]
remove the lbfgs accuracy requirement but leave the test. we need it for closure testing with lbfgs

**Comment by Borda** [comment]
See #1678

**Comment by tgaddair** [comment]
Hey @Borda, ready to land?

**Comment by Borda** [comment]
> Hey @Borda, ready to land?

yes, sure, just need to collect other approvals from @PyTorchLightning/core-contributors or @williamFalcon

**Pull request merged**